src/extension/index.ts

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { IArraySplice, observe } from 'mobx';
+import { Log } from 'sarif';
+import { CancellationToken, commands, DiagnosticSeverity, ExtensionContext, languages, Range, TextDocument, ThemeColor, Uri, window, workspace } from 'vscode';
+import { mapDistinct, parseRegion } from '../shared';
+import '../shared/extension';
+import { Baser } from './baser';
+import { loadLogs } from './loadLogs';
+import { Panel } from './panel';
+import { regionToSelection } from './regionToSelection';
+import { ResultDiagnostic } from './resultDiagnostic';
+import { Store } from './store';
+import { update, updateChannelConfigSection } from './update';
+
+export async function activate(context: ExtensionContext) {
+    const disposables = context.subscriptions;
+    Store.extensionPath = context.extensionPath;
+    Store.globalState = context.globalState;
+    disposables.push(commands.registerCommand('sarif.clearState', () => {
+        context.globalState.update('view', undefined);
+        commands.executeCommand('workbench.action.reloadWindow');
+    }));
+    const store = new Store();
+
+    // Basing
+    const urisNonSarif = await workspace.findFiles('**/*', '.sarif'); // Ignore folders?
+    const fileAndUris = urisNonSarif.map(uri => [uri.path.split('/').pop(), uri.path]) as [string, string][];
+    const baser = new Baser(mapDistinct(fileAndUris), store);
+
+    // Panel
+    const panel = new Panel(context, baser, store);
+    disposables.push(commands.registerCommand('sarif.showPanel', () => panel.show()));
+
+    // Diagnostics
+    const diagsAll = languages.createDiagnosticCollection('SARIF');
+    const setDiags = (doc: TextDocument) => {
+        if (doc.fileName.endsWith('.git')) return;
```

Comment:
>doc.fileName.endsWith('.git') [](start = 12, length = 29)

can potentially do the same with files in the .git folder and the .vscode folder

Reply:
Could you elaborate? I was only seeing spurious onDidOpen/onDidClose events for the '*.git' files so I needed to filter those out. I was not seeing any issues related to the '.git' folder and '.vscode' folder.

---
In reply to: [441201549](https://github.com/microsoft/sarif-vscode-extension/pull/282#discussion_r441201549) [](ancestors = 441201549)

Reply:
ahh okay. I was just thinking there's probably no need to take any action on files opened from the .vscode and .git folder in case those were opened, but that probably won't happen that often. I didn't know there were spurious "*.git" files being "opened".

---
In reply to: [441207206](https://github.com/microsoft/sarif-vscode-extension/pull/282#discussion_r441207206) [](ancestors = 441207206,441201549)

Reply:
Add a comment: "When the user opens a doc, VS Code commonly silently opens the associate `*.git`. We are not interested in these events."

Not:, while it's uncommon to open files from '.vscode' and '.git', it's a legitimate scenario.

---
In reply to: [442380218](https://github.com/microsoft/sarif-vscode-extension/pull/282#discussion_r442380218) [](ancestors = 442380218,441207206,441201549)